what4/src/What4/Utils/Process.hs

```diff
@@ -93,6 +93,7 @@ startProcess path args mcwd =
               , std_err = CreatePipe
               , create_group = False
               , cwd = mcwd
+              , delegate_ctlc = True
```

Comment:
what effect does this end up having?

Reply:
It's not directly related to the timeouts themselves, but it's relative to the more general handling of solvers and solver exceptions that is a part of this PR, and it additionally addresses the user Ctl-C usage.

The effect of this change is that the Ctl-C is now passed on to the solver process to allow that process to abort or do what the solver would normally do on a Ctl-C.  The solver management code should now handle the resulting solver's actions appropriately.   That said, this change could end up with other unintended consequences not revealed during testing, in which case it's easy to revert this single setting and it doesn't invalidate the rest of this PR.